xcm/procedural/src/v1/multilocation.rs

```diff
@@ -184,16 +184,18 @@ fn generate_conversion_from_v0() -> TokenStream {
 					Ok(MultiLocation::ancestor(#num_ancestors)),
 				#intermediate_match_arms
 				crate::v0::MultiLocation::#variant( #(#idents),* ) =>
-					Ok( #variant( #( #idents.try_into()? ),* ).into() ),
+					Ok( #variant( #( core::convert::TryInto::try_into(#idents)? ),* ).into() ),
 			}
 		})
 		.collect::<TokenStream>();
 
 	quote! {
-		impl TryFrom<crate::v0::MultiLocation> for MultiLocation {
+		impl core::convert::TryFrom<crate::v0::MultiLocation> for MultiLocation {
 			type Error = ();
-			fn try_from(v0: crate::v0::MultiLocation) -> core::result::Result<Self, ()> {
+			fn try_from(mut v0: crate::v0::MultiLocation) -> core::result::Result<Self, ()> {
 				use Junctions::*;
+
+				v0.canonicalize();
```

Comment:
This wasn't called before?

Reply:
Yeah it wasn't, even before the migration to the proc macro.

Reply:
What I should have asked, do you added this by intention? :D

Reply:
Yes, I thought that the conversion should always try and canonicalize the `MultiLocation` first, since it's an infallible operation. However, it does change the internal representation of the `MultiLocation` irreversibly, but I think that's ok because the non-canonical representation would've been "erroneous" anyway.